pkg/controller/monitor/monitor_controller.go

```diff
@@ -0,0 +1,198 @@
+		return fmt.Errorf("monitor-controller failed to watch ServiceMonitor elasticsearch-metrics resource: %w", err)
+	}
+
+	if err = utils.AddPodMonitorWatch(c); err != nil {
+		return fmt.Errorf("monitor-controller failed to watch PodMonitor resource: %w", err)
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileMonitor implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileMonitor{}
+
+type ReconcileMonitor struct {
+	client   client.Client
+	scheme   *runtime.Scheme
+	provider operatorv1.Provider
+	status   status.StatusManager
+}
+
+func getMonitor(ctx context.Context, cli client.Client) (*operatorv1.Monitor, error) {
+	instance := &operatorv1.Monitor{}
+	err := cli.Get(ctx, utils.DefaultTSEEInstanceKey, instance)
+	if err != nil {
+		return nil, err
+	}
+
+	return instance, nil
+}
+
+func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Monitor")
+
+	instance, err := getMonitor(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		r.status.SetDegraded("Failed to query Monitor", err.Error())
+		return reconcile.Result{}, err
+	}
+	reqLogger.V(2).Info("Loaded config", "config", instance)
+	r.status.OnCRFound()
+
+	variant, install, err := utils.GetInstallation(context.Background(), r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded("Installation not found", err.Error())
+			return reconcile.Result{}, err
+		}
+		r.status.SetDegraded("Failed to query Installation", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	pullSecrets, err := utils.GetNetworkingPullSecrets(install, r.client)
+	if err != nil {
+		reqLogger.Error(err, "error retrieving pull secrets")
+		r.status.SetDegraded("An error occurring while retrieving the pull secrets", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Create a component handler to manage the rendered component.
+	hdler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
+
+	// render prometheus components
+	component := render.Monitor(install, pullSecrets)
+
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, component); err != nil {
+		reqLogger.Error(err, "Error with images from ImageSet")
+		r.status.SetDegraded("Error with images from ImageSet", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	if err := hdler.CreateOrUpdateOrDelete(ctx, component, r.status); err != nil {
+		reqLogger.Error(err, err.Error())
+		r.status.SetDegraded("Error creating / updating resource", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Tell the status manager that we're ready to monitor the resources we've told it about and receive statuses.
+	r.status.ReadyToMonitor()
+
+	r.status.ClearDegraded()
+
+	if !r.status.IsAvailable() {
```

Comment:
From the tigerastatus, monitor will be available after the system is stablised.

```bash
$ kubectl get tigerastatus
NAME                  AVAILABLE   PROGRESSING   DEGRADED   SINCE
apiserver             True        False         False      27m
calico                True        False         False      27m
compliance            True        False         False      24m
intrusion-detection   True        False         False      24m
log-collector         True        False         False      24m
log-storage           True        False         False      24m
manager               True        False         False      24m
monitor               True        False         False      4m10s
```